options/options_parser.cc

```diff
@@ -587,6 +587,17 @@ bool AreEqualOptions(
     case OptionType::kInfoLogLevel:
       return (*reinterpret_cast<const InfoLogLevel*>(offset1) ==
               *reinterpret_cast<const InfoLogLevel*>(offset2));
+    case OptionType::kCompactionOptionsFIFO: {
+      CompactionOptionsFIFO lhs =
+          *reinterpret_cast<const CompactionOptionsFIFO*>(offset1);
+      CompactionOptionsFIFO rhs =
+          *reinterpret_cast<const CompactionOptionsFIFO*>(offset2);
+      if (lhs.max_table_files_size == rhs.max_table_files_size &&
```

Comment:
Yes, that is my ultimate aim, and hence I introduced functions like ParseStruct,  SerializeStruct and GetStringFromStruct. 

If you think about it, all options, including the high level DBOptions and CFOptions are nothing but nested structs. GetStringFromDBOptions and GetStringFromColumnFamilyOptions are now nothing but GetStringFromStruct<T> with a different name (I couldn't just remove GetStringFromDBOptions and GetStringFromColumnFamilyOptions, because they are in the public interface).  But currently the options parsing/serializing code is somewhat convoluted, and it would take many iterations to get to the ultimate goal of much cleaner generic options-related code.

Reply:
too bad c++ don't have reflection like Java or PHP :p